.github/workflows/build-test.yml

```diff
@@ -31,44 +31,15 @@ jobs:
       - name: Cache Coursier cache
         uses: coursier/cache-action@v6.4.0
 
-      - name: Set up JDK 17
-        uses: coursier/setup-action@v1.3.0
-        with:
-          jvm: temurin:1.17.0
-
-      - name: Code style check and binary-compatibility check
-        run: |-
-          cp .jvmopts-ci .jvmopts
-          sbt scalafmtCheckAll scalafmtSbtCheck headerCheck
-
-  compile-with-scala-2_12:
-    name: Compile with Scala 2.12
-    runs-on: ubuntu-22.04
-    if: github.repository == 'akka/akka-persistence-r2dbc'
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3.1.0
-        with:
-          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
-          fetch-depth: 0
-      - name: Checkout GitHub merge
-        if: github.event.pull_request
-        run: |-
-          git fetch origin pull/${{ github.event.pull_request.number }}/merge:scratch
-          git checkout scratch
-
-      - name: Cache Coursier cache
-        uses: coursier/cache-action@v6.4.0
-
       - name: Set up JDK 8
         uses: coursier/setup-action@v1.3.0
         with:
           jvm: temurin:1.8.0
 
-      - name: Code style check and binary-compatibility check
+      - name: Code style check, compilation and binary-compatibility check
         run: |-
           cp .jvmopts-ci .jvmopts
-          sbt ++2.12 compile Test/compile
+          sbt scalafmtCheckAll scalafmtSbtCheck headerCheck mimaReportBinaryIssues +Test/compile
```

Comment:
The diff might be confusing. Removed the 2.12 compile job, and instead `+Test/compile` in the check job